spoti-friends/src/friendActivity/views/ListeningActivityDetails.swift

```diff
@@ -33,28 +39,47 @@ struct ListeningActivityDetails: View {
         VStack(alignment: .leading, spacing: 2) {
             // Username row
             HStack {
-                Text(displayName)
-                    .fontWeight(.medium)
+                Link(destination: URL(string: profile.spotifyUri)!) {
+                    Text(profile.displayName)
+                        .fontWeight(.medium)
+                }
                 Spacer()
                 TrackTimestampView(nowPlaying: currentTrack.playedWithinLastFifteenMinutes,
                                    timestamp: currentTrack.timestamp)
             }
             
             // Song details row
             HStack {
-                Text(currentTrack.track?.name ?? "Error")
-                    .lineLimit(1)
+                Link(destination: URL(string: trackDetails.spotifyUri)!) {
+                    Text(trackDetails.name)
+                        .lineLimit(1)
+                }
                 Text("•")
-                Text(currentTrack.track?.artist?.name ?? "Error")
-                    .lineLimit(1)
+                Link(destination: URL(string: artist.spotifyUri)!) {
+                    Text(artist.name)
+                        .lineLimit(1)
+                }
             }
             
             // Context details row
-            HStack {
-                contextIcon
-                    .padding(.trailing, -6)
-                Text(currentTrack.track?.context?.name ?? "Error")
-                    .lineLimit(1)
+            // Wrap the row with a link to the context item if it is not nil.
+            if let context = trackDetails.context {
```

Comment:
Instead of `if let`, also set this as a `@State` variable, so that we don't need a trivial `if-else` statement